module/observations/helper.js

```diff
@@ -2819,20 +2823,53 @@ module.exports = class ObservationsHelper {
           let topLevelEntityId = bodyData[topLevelEntityType];
 
 
-          let roles = bodyData.role.split(",");
+          let tenantDetails = await userService.fetchPublicTenantDetails(tenantData.tenantId);
+          if (!tenantDetails.data || !tenantDetails.data.meta || tenantDetails.success !== true) {
+            throw {
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.FAILED_TO_FETCH_TENANT_DETAILS,
+            };
+          }
+
+          let observableEntityKeys = tenantDetails.data.meta.observableEntityKeys;
+
+          if(observableEntityKeys === undefined || observableEntityKeys === null || observableEntityKeys.length === 0){
+            throw {
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.OBSERVABLE_ENTITY_KEYS_NOT_FOUND,
+            };
+          }
+          
+          let KyetoValidate = [];
+
+          for(let i=0;i<observableEntityKeys.length;i++){
```

Comment:
@borkarsaish65 please correct the indendations and format the newly written code using prettier or eslint

Reply:
done